### PR TITLE
🚑️ add more logging to GC

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

While trying to reproduce the error from https://github.com/ITISFoundation/osparc-simcore/issues/3975 locally, I had troubles doing so.

Extending GC to display some more meaningful logs.

- ♻️ `log_decorator` can now has an additional parameter `log_traceback` to display tracebacks.
- 🚑️ `_remove_single_orphaned_service` will now display a proper traceback

**Bonus**: fixed `log_decorator` when applied to not async function. It was not working as expected

## Related issue/s

- https://github.com/ITISFoundation/osparc-simcore/issues/3975

## Checklist

- [x] Unit tests for the changes exist

### CODE REVIEW

File: packages/service-library/src/servicelib/logging_utils.py [hunk 1]
```diff
@@ -183,8 +186,11 @@ def log_decorator_wrapper(*args, **kwargs):
                     )
                 except:
                     # log exception if occurs in function
-                    logger_obj.error(
-                        "Exception: %s", sys.exc_info()[1], extra=extra_args
+                    logger_obj.exception(
```
Comment (sanderegg) [inline packages/service-library/src/servicelib/logging_utils.py:189]: I though log.exception already automatically prints the traceback?

File: packages/service-library/src/servicelib/logging_utils.py [hunk 2]
```diff
@@ -183,8 +186,11 @@ def log_decorator_wrapper(*args, **kwargs):
...
+                        "Exception: %s",
+                        sys.exc_info()[1],
+                        extra=extra_args,
```
Comment (sanderegg) [inline packages/service-library/src/servicelib/logging_utils.py:192]: if you change the logger to use exception, please do so also in the case above...

**Review by sanderegg — APPROVED** [review]

**Review by sanderegg — APPROVED** [review]

File: packages/service-library/src/servicelib/logging_utils.py (hunk 1 above)
Comment (pcrespov) [reply packages/service-library/src/servicelib/logging_utils.py:189]: Yes. I think this is equivalent to ``logger_obj.exception("Exception:")``. Perhaps you can add some extra info on the msg and not just "Exception"

Comment (GitHK) [reply packages/service-library/src/servicelib/logging_utils.py]: My bad forgot to revert this. Also added tests for this part.

File: packages/service-library/src/servicelib/logging_utils.py (hunk 2 above)
Comment (GitHK) [reply packages/service-library/src/servicelib/logging_utils.py:192]: This is not what I intended to do.

**Pull request merged**